numpy/core/tests/test_scalarprint.py

```diff
@@ -45,6 +46,21 @@ def check(v):
         check(1e15)
         check(1e16)
 
+    def test_py2_float_print(self):
+        # gh-10753
+        # In python2, the python float type implements an obsolte method
+        # tp_print, which overrides tp_repr and tp_str when using the "print"
+        # keyword/method to output to a "real file" (ie, not a StringIO). Make
+        # sure we don't inherit it.
+        x = np.double(0.1999999999999)
+        f = tempfile.TemporaryFile('r+t') # must output to real file, not StringIO
+        print(x, file=f)
+        f.seek(0)
+        output = f.read()
+        f.close()
+        assert_equal(output, '0.1999999999999\n')
```

Comment:
Perhaps assert equality with `str(x) + '\n'`? Not sure if there ever can be rounding issues here, but what we care about is that `str` and `print` give the same result...